Algorithms/Implementation/Lisa's Workbook/solution.cpp

```diff
@@ -0,0 +1,55 @@
+#include <bits/stdc++.h>
+
+using namespace std;
+
+int workbook(int n, int k, vector<int> arr) {
```

Comment:
I guess the problem was with minor constraints. Let's see if we can improve the solution with Maths.
Looks good to me for the constraints.